File: api/src/Entity/VoirRessource.php
<?php

namespace App\Entity;

use App\Repository\VoirRessourceRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Attribute\Groups;

#[ORM\Entity(repositoryClass: VoirRessourceRepository::class)]
class VoirRessource
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'voirRessources')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Utilisateur $Utilisateur = null;

    #[ORM\ManyToOne(inversedBy: 'voirRessources')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Ressource $Ressource = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUtilisateur(): ?Utilisateur
    {
        return $this->Utilisateur;
    }

    public function setUtilisateur(?Utilisateur $Utilisateur): static
    {
        $this->Utilisateur = $Utilisateur;

        return $this;
    }

    public function getRessource(): ?Ressource
    {
        return $this->Ressource;
    }

    public function setRessource(?Ressource $Ressource): static
    {
        $this->Ressource = $Ressource;

        return $this;
    }
}

File: mobile/ressources_re_mobile/lib/classes/VoirRessource.dart
// ignore_for_file: file_names

import 'package:ressources_re_mobile/classes/Ressource.dart';
import 'package:ressources_re_mobile/classes/Utilisateur.dart';

class VoirRessource {
  int? id;
  Utilisateur? utilisateur;
  Ressource? ressource;

  VoirRessource();

  int? getId() {
    return id;
  }

  Utilisateur? getUtilisateur() {
    return utilisateur;
  }

  void setUtilisateur(Utilisateur? utilisateur) {
    this.utilisateur = utilisateur;
  }

  Ressource? getRessource() {
    return ressource;
  }

  void setRessource(Ressource? ressource) {
    this.ressource = ressource;
  }

  VoirRessource.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    utilisateur = json['utilisateur'];
    ressource = json['ressource'];
  }
}
